src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,35 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                let check_lock_res =
+                    self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock));
+                if let Err(e) = check_lock_res {
+                    *can_be_cached.unwrap() = can_be_cached_by_lock;
```

Comment:
`can_be_cached` can be `None`. Maybe:

```rust
if !can_be_cached_by_lock {
    *can_be_checked.unwrap() = false;   // the unwrap here is fine, since `can_be_cached_by_lock` can only be changed when `can_be_cached` is Some due to my previous comment
}
check_lock_res?;
```